src/platform/baytrail/dai.c

```diff
@@ -12,108 +13,86 @@
 #include <sof/ssp.h>
 #include <ipc/dai.h>
 #include <ipc/stream.h>
+#include <ipc/topology.h>
 #include <config.h>
 
-static struct dai ssp[] = {
```

Comment:
Could you elaborate on why dynamic allocation is better than static in this case? You still need to hardcode the number of SSPs and manually assign address and irq.